### PR TITLE
HTML Documentation Framework

### PR DESCRIPTION
**Pull request by ximinez**
This is a continuation of #1857 and #1864.

The only thing that's really new is the CMake `docs` target.

### CODE REVIEW

**Comment by mellery451** [comment]
👍

File: docs/main.qbk [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+]
+
+[library rippled
+    [quickbook 1.6]
+    [copyright 2012 - 2016 Ripple Labs Inc.]
+    [purpose C++ Library]
+    [license
+        Distributed under the ISC License
+    ]
+    [authors [Labs, Ripple]]
```
Comment (seelabs) [inline docs/main.qbk]: `Labs, Ripple` seems odd. Is that correct?

File: docs/makeqbk.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (seelabs) [inline docs/makeqbk.sh]: If this works on windows I have no problem. I thought some windows shell had bash in `/usr/bin`.

Comment (seelabs) [inline docs/makeqbk.sh]: Making a note that @ximinez Checked both cygwin and the git shell. This shebang works.

File: docs/main.qbk (hunk 1 above)
Comment (ximinez) [inline docs/main.qbk]: Good question. @vinniefalco?

**Comment by ximinez** [comment]
Merged to develop as f27348c, 96b1774, and 7d46d15.

**Pull request closed without merging**